src/main/java/hudson/plugins/active_directory/AbstractActiveDirectoryAuthenticationProvider.java

```diff
@@ -61,5 +62,6 @@ protected void additionalAuthenticationChecks(UserDetails userDetails, UsernameP
     /**
      * Setting this to true might help with diagnosing login problem.
      */
+    @SuppressFBWarnings(value = "MS_SHOULD_BE_FINAL", justification = "It should, but changing would break backwards compatibility.")
```

Comment:
I don't agree that this "should" be final. These diagnostic fields are left mutable so that groovy console can be used to dynamically turn/off probes.